R/render.R

```diff
@@ -280,6 +290,42 @@ render_quarto_lang <- function(language_code, path, output_dir, type) {
 
 }
 
+#' Filter the freeze directory
+#'
+#' This function filters the freeze directory keeping only the files
+#' that match the language code.
+#'
+#' @param temporary_directory Temporary directory where the project is
+#' @param project_name Name of the project directory
+#' @param language_code The Language code for the current rendering
+#' @dev
+filter_freeze_directory <- function(temporary_directory,
+                                    project_name,
+                                    language_code) {
+  freeze_path <- fs::path(temporary_directory, project_name, "_freeze")
+  freeze_temp <- fs::path(
+    temporary_directory,
+    project_name, paste0("_freeze.", language_code)
+  )
+  freeze_ls <- fs::dir_ls(freeze_path, recurse = TRUE)
+
+  freeze_lang <- purrr::keep(
+    freeze_ls, \(x) grepl(paste0("\\.", language_code, "$"), x)
```

Comment:
so it magically works without passing the language code, interesting (I really had no idea)

Reply:
Ah R's scoping rules 🙈